chord/helpers.go

```diff
@@ -1,17 +1,39 @@
 package chord
 
-import "sync"
+import (
+	"crypto"
+	"encoding/hex"
+	"sync"
+)
 import . "github.com/dosarudaniel/CS438_Project/services/chord_service"
 
 // In this file, we put all helper functions and types used by Chord package
 // "helper" can be defined as something that is not worth own file/package and has tiny role in the chord functionality
 
+type ipAddr string
+
 type nodeWithMux struct {
 	node *Node
 	sync.RWMutex
 }
 
 type successorsListWithMux struct {
-	successorsList []Node
+	list []Node
+	sync.RWMutex
+}
+
+type ipToStubMap map[ipAddr]ChordClient
+type stubsPoolWithMux struct {
+	pool ipToStubMap
 	sync.RWMutex
 }
+
+func hashString(s string) (string, error) {
+	sha256 := crypto.SHA256.New()
+	_, err := sha256.Write([]byte(s))
+	if err != nil {
+		return "", err
+	}
+	hash := sha256.Sum(nil)
+	return hex.EncodeToString(hash), nil
```

Comment:
We might need to keep only `m` bits

Reply:
> In my opinion we should close the opened connections after one RPC was executed - it's easy to implement this for the moment (We can analyze the impact of this).
`getStubFor` already works, so why bother